notebook/static/tree/js/notebooklist.js

```diff
@@ -835,6 +866,7 @@ define([
         // Add in the date that the file was last modified
         item.find(".item_modified").text(utils.format_datetime(model.last_modified));
         item.find(".item_modified").attr("title", moment(model.last_modified).format("YYYY-MM-DD HH:mm"));
+        item.find(".file_size").html(utils.format_filesize(model.size) || "&nbsp;");
```

Comment:
the use of `.html()` should be a red flag. From a security standpoint it would not pass an audit. We had security issues before because of HTML usage. 

If somewhere where to replace you API to return `cannot find <size> for file <file>` the you get scrip injection in your page, and your client is compromised. 

If you've seen use of `.html()` in other place of the javascript where `.text()` should/could be used, please open an issue.

Reply:
Good question, there are several way to approach that. A simple one would be to use `.html()` just for `&nbsp;`:

```
if (cfoo){
	$(...).text(foo); 
} else {
    $(...).html("&nbsp");
}
```

That reduce the use of html so it's one less risk. 

Thinking about it more, `&nbsp;` is also just an escape sequence for unicode character [Non Breaking Space](http://www.fileformat.info/info/unicode/char/00a0/index.htm), which you can enter with your keyboard using some clever keyboard combinaison `| |` (nbsp), not like `| |`(space), check the source, on mac it's Alt Space. That's 1) annoying to type 2) annoying to see when reading code. That's not optimal. After thinking a few minutes more you can realize that hopefully most programming language allow you to have escape sequence, knowing that Non-Breaking space is unicode [`U+00A0`](http://www.fileformat.info/info/unicode/char/00a0/index.htm) you can go with : 
```
.text(checked===0 ? '\xA0' : checked); 
```

If you want to even avoid using jQuery you [can use](https://caniuse.com/#feat=innertext) the `innerText` attribute and assign to it.

With your gained knowledge you can even open an issue for `$('#counter-select-all').html(checked===0 ? '&nbsp;' : checked);` to be fixed. 

As an exercise – if you like the fun of hacking – get in a shoes of an attacker and try to modify the server side to not return the size as an integer but a string, and attempt to have a script injection that display a message in the frontend.

Reply:
Stepping back a bit from unicode discussions: do we actually need a non-breaking space at all? I think the layout should still work if the element has no text.

Reply:
I fixed it with using `\xA0`. Without text, the `span` element wrapping it doesn't seem to adhere to the width it's given.